### DISCUSSION TITLE
taro构建的qq小程序在部分安卓机上卡顿严重

### DISCUSSION THREAD

**Discussion by kangling098** [NervJS/taro#10122] [Q & A]
taro版本: 3.3.3
构建的qq小程序,使用
"dev:qq": "cross-env BUILD_ENV=development npm run build:qq -- --watch",
"dev:terser:qq": "cross-env NODE_ENV=production BUILD_ENV=development npm run build:qq -- --watch --terser",

这两种模式构建的小程序,使用红米k30、华为p20、vivo X30等安卓机型,在开发者工具上使用预览模式时,运行流畅,但是在构建生产包
"build:qq": "taro build --type qq"
在上述机型上出现显著卡顿,点击响应延迟、页面加载延迟等问题

以上问题,在ios设备上不会出现,体验流畅

问题: qq小程序构建生产包的时候,是否由于兼容性代码,导致应用性能出现问题

**Comment by Chen-jj** [comment]
使用 `<CustomWrapper>` 试试

https://docs.taro.zone/blog/2021-02-08-taro-jxpp

**Reply by kangling098** [reply]
em...我们的项目中很多地方都使用了 CustomWrapper来进行优化,但是还是会有这个问题,而且只有构建生产包的时候,才会出现上述的问题

**Reply by Chen-jj** [reply]
先按文章说方法都处理一下，如果还有问题，就要提供一下 Demo 了，性能问题挺难定位的。

**Reply by kangling098** [reply]
em,艰难
